File: LeetCode Solutions/LeetCoding Challenge/January 2021 LeetCoding Challenge/Max Number of K-Sum Pairs/Max Number of K-Sum Pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        nums.sort()
        numberOfOperations, sum, i, j = 0, 0, 0, len(nums)-1
        while (i < j):
            sum = nums[i] + nums[j]
            if (sum == k):
                i += 1
                j -= 1
                numberOfOperations += 1
            elif (sum < k):
                i += 1
            else:
                j -= 1
        return numberOfOperations

File: LeetCode Solutions/LeetCoding Challenge/January 2021 LeetCoding Challenge/Max Number of K-Sum Pairs/Max Number of K-Sum Pairs.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int numberOfOperations = 0, sum = 0, i = 0, j = nums.size()-1;
        while (i < j){
            if (nums[i] + nums[j] == k){
                i++;
                j--;
                numberOfOperations++;
            }
            else if (nums[i] + nums[j] < k){
                i++;
            } else{
                j--;
            }
        }
        return numberOfOperations;
    }
};